src/main/java/com/revaturelabs/ask/user/UserServices.java

```diff
@@ -0,0 +1,122 @@
+package com.revaturelabs.ask.user;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import com.revaturelabs.ask.user.UserRepo;
+
+@Service
+public class UserServices {
+
+  @Autowired
+  private UserRepo userRepo;
+
+  /**
+   * "findAll" basically gets all
+   * the users from the H2 database.
+   * 
+   * @return List of all users in database.
+   */
+  public List<User> findAll() {
+    return (List<User>) userRepo.findAll();
+  }
+
+  /**
+   * "findById" finds the user by their ID.
+   * 
+   * @param id
+   * @return
+   * @throws UserNotFoundException
+   */
+  public User findById(int id) throws UserNotFoundException {
+    Optional<User> user = userRepo.findById(id);
+
+    if (!user.isPresent()) {
+      throw new UserNotFoundException();
+    }
+
+    return user.get();
+  }
+
+  /**
+   * "create" gets the user object from the API and 
+   * saves it into the database.
+   * 
+   * @param user
+   * @return
+   */
+  public User create(User user) {
+    return userRepo.save(user);
+  }
+
+  /**
+   * "update" checks if the user exists beforehand.
+   * If so, it saves changes into the specified row
+   * of the database, given it doesn't overwrite the ID.
+   * 
+   * If not, it throws the exception.
+   * 
+   * @param user
+   * @return
+   * @throws UserNotFoundException
+   * @throws UserConflictException
+   */
+  public User update(User user) throws UserNotFoundException, UserConflictException {
+    Optional<User> existingUser = userRepo.findById(user.getId());
+
+    User updatedUser = null;
+    if (existingUser.isPresent()) {
+      try {
+        updatedUser = userRepo.save(user);
+      } catch (DataIntegrityViolationException e) {
+        throw new UserConflictException();
```

Comment:
UserConflictException is for Users already existing in the database.  DataIntegrityViolationException is for any constraint violation, not just PK-related violations.  Fix the messaging